go-controller/pkg/ovn/gatewayrouter/policybasedroutes.go

```diff
@@ -0,0 +1,209 @@
+	netInfo           util.NetInfo
+	nbClient          client.Client
+}
+
+func NewPolicyBasedRoutesManager(nbClient client.Client, clusterRouterName string, netInfo util.NetInfo) *PolicyBasedRoutesManager {
+	return &PolicyBasedRoutesManager{
+		clusterRouterName: clusterRouterName,
+		netInfo:           netInfo,
+		nbClient:          nbClient,
+	}
+}
+
+func (pbr *PolicyBasedRoutesManager) Add(nodeName, mgmtPortIP string, hostIfCIDR *net.IPNet, otherHostAddrs []string) error {
+	var l3Prefix string
+	if utilnet.IsIPv6(hostIfCIDR.IP) {
+		l3Prefix = "ip6"
+	} else {
+		l3Prefix = "ip4"
+	}
+
+	matches := sets.New[string]()
+	for _, hostIP := range append(otherHostAddrs, hostIfCIDR.IP.String()) {
+		// embed nodeName as comment so that it is easier to delete these rules later on.
+		// logical router policy doesn't support external_ids to stash metadata
+		networkScopedSwitchName := pbr.netInfo.GetNetworkScopedSwitchName(nodeName)
+		matchStr := fmt.Sprintf(`inport == "%s%s" && %s.dst == %s /* %s */`,
+			ovntypes.RouterToSwitchPrefix, networkScopedSwitchName, l3Prefix, hostIP, networkScopedSwitchName)
+		matches = matches.Insert(matchStr)
+	}
+
+	if err := pbr.sync(
+		nodeName,
+		matches,
+		ovntypes.NodeSubnetPolicyPriority,
+		mgmtPortIP,
+	); err != nil {
+		return fmt.Errorf("unable to sync node subnet policies, err: %v", err)
+	}
+
+	return nil
+}
+
+// This function syncs logical router policies given various criteria
+// This function compares the following ovn-nbctl output:
+
+// either
+
+// 		72db5e49-0949-4d00-93e3-fe94442dd861,ip4.src == 10.244.0.2 && ip4.dst == 172.18.0.2 /* ovn-worker2 */,169.254.0.1
+// 		6465e223-053c-4c74-a5f0-5f058c9f7a3e,ip4.src == 10.244.2.2 && ip4.dst == 172.18.0.3 /* ovn-worker */,169.254.0.1
+// 		7debdcc6-ad5e-4825-9978-74bfbf8b7c27,ip4.src == 10.244.1.2 && ip4.dst == 172.18.0.4 /* ovn-control-plane */,169.254.0.1
+
+// or
+
+// 		c20ac671-704a-428a-a32b-44da2eec8456,"inport == ""rtos-ovn-worker2"" && ip4.dst == 172.18.0.2 /* ovn-worker2 */",10.244.0.2
+// 		be7c8b53-f8ac-4051-b8f1-bfdb007d0956,"inport == ""rtos-ovn-worker"" && ip4.dst == 172.18.0.3 /* ovn-worker */",10.244.2.2
+// 		fa8cf55d-a96c-4a53-9bf2-1c1fb1bc7a42,"inport == ""rtos-ovn-control-plane"" && ip4.dst == 172.18.0.4 /* ovn-control-plane */",10.244.1.2
+
+// or
+
+// 		822ab242-cce5-47b2-9c6f-f025f47e766a,ip4.src == 10.244.2.2  && ip4.dst != 10.244.0.0/16 /* inter-ovn-worker */,169.254.0.1
+// 		a1b876f6-5ed4-4f88-b09c-7b4beed3b75f,ip4.src == 10.244.1.2  && ip4.dst != 10.244.0.0/16 /* inter-ovn-control-plane */,169.254.0.1
+// 		0f5af297-74c8-4551-b10e-afe3b74bb000,ip4.src == 10.244.0.2  && ip4.dst != 10.244.0.0/16 /* inter-ovn-worker2 */,169.254.0.1
+
+// The function checks to see if the mgmtPort IP has changed, or if match criteria has changed
+// and removes stale policies for a node for the NodeSubnetPolicy in SGW.
+// TODO: Fix the MGMTPortPolicy's and InterNodePolicy's ip4.src fields if the mgmtPort IP has changed in LGW.
```

Comment:
this todo still applicable?

Reply:
hmm who added this todo? :o

Reply:
... deep down ... 

[You](https://github.com/ovn-org/ovn-kubernetes/commit/62e0171bc6e9115947fa8d19e24cf7ef25993b25#diff-9c68f154bc9189eb61d17c17d1392ee36c13a497f4f742a98250c2e0a272d48cR826) know the answer. :laughing: